addons/binding/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/handler/NeeoRoomProtocol.java

```diff
@@ -0,0 +1,354 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.neeo.handler;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.neeo.NeeoBindingConstants;
+import org.openhab.binding.neeo.NeeoUtil;
+import org.openhab.binding.neeo.internal.models.ExecuteResult;
+import org.openhab.binding.neeo.internal.models.ExecuteStep;
+import org.openhab.binding.neeo.internal.models.NeeoRecipe;
+import org.openhab.binding.neeo.internal.models.NeeoRoom;
+import org.openhab.binding.neeo.internal.models.NeeoScenario;
+import org.openhab.binding.neeo.internal.models.NeeoScenarios;
+import org.openhab.binding.neeo.internal.net.HttpRequest;
+import org.openhab.binding.neeo.internal.type.UidUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This protocol class for a Neeo Room
+ *
+ * @author Tim Roberts - Initial contribution
+ */
+class NeeoRoomProtocol implements AutoCloseable {
+
+    /** The logger */
+    private final Logger logger = LoggerFactory.getLogger(NeeoRoomProtocol.class);
+
+    /** The {@link HttpRequest} to use for the room */
+    private final HttpRequest request;
```

Comment:
This seems unused

Reply:
Good catch - was used at one point but is no longer used